client/lib/pages/symptom_checker/symptom_checker_view.dart

```diff
@@ -0,0 +1,199 @@
+
+  const SymptomCheckerView({Key key, @required this.dataSource})
+      : super(key: key);
+
+  @override
+  _SymptomCheckerViewState createState() => _SymptomCheckerViewState();
+}
+
+class _SymptomCheckerViewState extends State<SymptomCheckerView>
+    implements SymptomCheckerPageDelegate {
+  final PageController _controller = PageController();
+  SymptomCheckerModel _model;
+  List<Widget> _pages;
+  bool inTransition = false;
+
+  @override
+  void didChangeDependencies() async {
+    super.didChangeDependencies();
+    await _initModel();
+    _modelChanged();
+  }
+
+  // Unlike other content-based widgets, the symptom checker
+  // will not dynamically react to new content.
+  Future<void> _initModel() async {
+    if (_model != null) {
+      return;
+    }
+    // TODO: Reduce this boilerplate
+    //Locale locale = Localizations.localeOf(context);
+    try {
+      final countryIsoCode =
+          (await UserPreferencesStore.readFromSharedPreferences())
+              .countryIsoCode;
+      final logicContext =
+          await LogicContext.generate(isoCountryCode: countryIsoCode);
+      final store = widget.dataSource;
+      await store.update();
+      var content = store.symptomChecker;
+      await Dialogs.showUpgradeDialogIfNeededFor(context, content);
+      _model = SymptomCheckerModel(content, logicContext);
+    } catch (err, stacktrace) {
+      print('Error loading content: $err\n$stacktrace');
+    }
+    _model.addListener(_modelChanged);
+    if (!mounted) return;
+    setState(() {});
+  }
+
+  Future<bool> _onWillPop() async {
+    // If the page controller hasn't been built yet then we're on the first page
+    if (_controller.hasClients == false) return true;
+
+    if (inTransition) return false;
+
+    if (_controller.page == 0) {
+      return true;
+    } else {
+      await goBack();
+      return false;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: _onWillPop,
+      child: Material(
+        color: Constants.backgroundColor,
+        child: Column(
+          children: <Widget>[
+            PageHeader(
+              inSliver: false,
+              title: 'Check-Up',
+              appBarColor: Constants.backgroundColor,
+              trailing: FlatButton(
+                padding: EdgeInsets.zero,
+                child: ThemedText(
+                  'Cancel',
+                  variant: TypographyVariant.button,
+                  style: TextStyle(color: Constants.whoBackgroundBlueColor),
+                ),
+                onPressed: () =>
+                    Navigator.of(context).popUntil((route) => route.isFirst),
+              ),
+            ),
+            Expanded(child: _buildPage(context)),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildPage(BuildContext context) {
+    if (_model == null) {
+      return _buildMessage('Loading...', loading: true);
+    }
+    if (_model.isFatalError) {
```

Comment:
Errors like this should be logged:

https://github.com/WorldHealthOrganization/app/blob/aef8359dcddd2b213af8aa1aaf7c4ab7ea704531/client/lib/components/carousel/carousel.dart#L95

@theswerd - is there a way that `FirebaseAnalytics` can log an error rather than just an event? I want it to explicitly be an error so that it's not lost.

Reply:
I'm not sure if we are curating a list of event names.  I'm adding one here for this:  `SymptomCheckerModelError`.